src/ray/gcs/redis_context.cc

```diff
@@ -598,55 +670,53 @@ Status RedisContext::Connect(const std::string &address,
 
   RAY_CHECK(!context_);
   RAY_CHECK(!redis_async_context_);
+
+  bool should_reconnect = true;
+
+  // Remember function arguments for reconnection
+  if (is_first_connect_) {
```

Comment:
We can have two Connect methods: first takes address, username etc and save them and the second has zero parameters and do the connection based on the saved ones. The first Connect can then just save and call the second Connect.